mindsdb/interfaces/chatbot/chatbot_task.py

```diff
@@ -145,21 +158,46 @@ def answer_to_user(self, chat_id):
             # the last message is from bot
             return
 
+        df = df.rename(columns={
+            text_col: 'text',
+            username_col: 'username',
+            time_col: 'sent_at'
+        })
+        model_output = self.apply_model(df, chat_id)
+
+        # send answer to user
+        ast_query = Insert(
+            table=Identifier(t_params['name']),
+            columns=[t_params['chat_id_col'], t_params['text_col']],
+            values=[
+                [chat_id, model_output],
+            ]
+        )
+
+        self.db_handler.query(ast_query)
+
+    def _chat_history_to_conversation(self, df, chat_id):
+
+        bot_username = self.params['bot_username']
+
         question_col = self.params['model']['user_column']
         answer_col = self.params['model']['bot_column']
 
+        history_from = self.chat_memory[chat_id].get('history_from')
```

Comment:
Just to double check my interpretation: this is retrieving all messages that have not been processed by the memory, yes?

Reply:
history_from is the line to split between conversations for current and previous mode. It was done to avoid interfering old messages (from previous mode) with current bot and mislead assistant